src/model/config.ts

```diff
@@ -0,0 +1,54 @@
+import BN from "bn.js";
+import { ConfigurableCorePool } from "../core/configurablePool";
+
+export class PoolConfig {
+    id: string;
+    tickSpacing: number;
+    token0: string;
+    token1: string;
+    fee: number;
+}
+
+export class Record {
+    id: string;
+    eventId: number;
+    actionName: string;
+    actionParams: object;
+    timestamp: Date;
+}
+
+export class Snapshot {
+    id: string;
+    nonce: number;
+    baseSnapshot: Snapshot;
+    poolConfig: PoolConfig;
+    records: Record[];
+    baseRecordId: string;
+    lastRecordId: string;
+    used: boolean;
+    timestamp: Date;
+    snapshotStorage: SnapshotStorage;
+
+    static from(id: string): Snapshot {
+        // TODO
+        return new Snapshot();
+    }
+
+    private persist(configurableCorePool: ConfigurableCorePool): string {
+        // TODO
+        return '0';
+    }
+
+}
+
+export class SnapshotStorage {
```

Comment:
I looked at the SQL table for Snapshot, maybe we should just flatten the `Snapshot` so it contains all the actual data currently stored in `SnapshotStorage`?

Reply:
It's also mixture problem mentioned above. looks like it's quite confusing. will think about it.